### PR TITLE
Task/prod data script/2380 - part 1

### PR DESCRIPTION
**Pull request by conbrad**
- Entrypoint code and tests for generating dates and times and iterating over the tif directory
- The `time.py` code _should_ live in a common time module since a lot of it overlaps with `time.py` in `api` but our project isn't set up for this (yet)
- Some of the code is also reused from `sfms` but that module is written in python2 since it runs on an old windows box
- `tileserv/tools/sfms_generator.py` is just skeleton and commenting around what will be hooked up next.
- Running the code will currently just list any `hfi**.tif{f}` files in the directory argument you supply.
    - Can be run with `python3 tif_pusher.py <issue_date> <for_date> <tif_dir>`  
- Part 2 will add more of the functionality.
# Test Links:
[Percentile Calculator](https://wps-pr-2451.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-2451.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2451.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2451.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2451.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-2451.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-2451.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-2451.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

File: tileserv/tools/sfms_generator.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+def generate_forecast(issue_date: date, for_date: date):
+    """ 
+        Forecasts will be generated multiple times for any given date.
+
+        Generally 3 to 4 forecasts will be generated for a single date
+        since that's what we've seen SFMS output
+
+        e.g. issue_date: 24
+                    - forecast: 25
```
Comment (andrea-williams) [inline tileserv/tools/sfms_generator.py:29]: Aside - is SFMS not running using morning forecast data for the date of interest? I.e., for issue_date=24, there should also be a forecast for 24?

File: tileserv/tools/time.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+def noon_time(issue_date: date):
+    """ Pin the time of an issue date to noon time, 13:00 """
+    return pin_issue_date_to_time(issue_date, time(13, 0, 0))
+
+
+def evening_time(issue_date: date):
+    """ Pin the time of an issue date to evening time, 16:45 """
+    return pin_issue_date_to_time(issue_date, time(16, 45, 0))
```
Comment (andrea-williams) [inline tileserv/tools/time.py:22]: According to my notes from our SFMS discussion with Devona in the summer, SFMS runs on forecast data at 8 am and 5:45 pm, should this should be 17:45

File: tileserv/tools/sfms_generator.py (hunk 1 above)
Comment (conbrad) [reply tileserv/tools/sfms_generator.py:29]: No idea, that's why I left these as comments and put it up for an early review. I suspect you know more than I do.

Comment (conbrad) [reply tileserv/tools/sfms_generator.py:29]: Looking at the data, yes it looks like what you stated is the case

File: tileserv/tools/time.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    return pin_issue_date_to_time(issue_date, time(8, 0, 0))
+
+
+def noon_time(issue_date: date):
+    """ Pin the time of an issue date to noon time, 13:00 """
+    return pin_issue_date_to_time(issue_date, time(13, 0, 0))
+
+
+def evening_time(issue_date: date):
+    """ Pin the time of an issue date to evening time, 16:45 """
```
Comment (andrea-williams) [inline tileserv/tools/time.py:21]:
```suggestion
    """ Pin the time of an issue date to evening time, 17:45 """
```

Comment (andrea-williams) [reply tileserv/tools/time.py:21]: atm I don't see where/how these functions are going to be used. tif_pusher.py seems to only use an issue _date_, not a time. I'm presuming that comes up in a future PR?

**Review by andrea-williams — APPROVED** [review]

File: tileserv/tools/time.py (hunk 3 above)
Comment (conbrad) [reply tileserv/tools/time.py:21]: Yeah tif pusher will decorate the input date with times for uploading to API and ultimately end up as `run_datetime`

**Review by dgboss — APPROVED** [review]

**Comment by conbrad** [comment]
Closing as this is not the way

**Pull request closed without merging**